allennlp/data/dataset_readers/vqav2.py

```diff
@@ -534,31 +533,17 @@ def apply_token_indexers(self, instance: Instance) -> None:
     def _get_answers_by_question_id(self, split):
```

Comment:
Did you mean to decorate this with `lru_cache` as well?

Reply:
No, though I could see it. In a normal run, that function only gets called twice, and with different arguments. But with our way of using the validation set twice, we could save 15 seconds, I guess.

Reply:
15 seconds per epoch, that is

Reply:
This gets called each epoch (since we're doing lazy loading). And takes about 3 minutes to run each time. It's definitely worth caching.

Reply:
Why does it take so much time? It's 15 seconds on my Mac.

Reply:
Whaaat? How is it so quick for you? It consistently takes 2.5 - 3 minutes on every computer I've ran it on. See these beaker logs for instance:

<img width="1126" alt="image" src="https://user-images.githubusercontent.com/8812459/102117074-b6039480-3df2-11eb-82ca-2c33eeeac3de.png">

Reply:
Are you using a different config? Different dataset?

Reply:
This is the config: https://data.beaker.org/datasets/01erwyaezkf7csjmyj8hkavm0q/files/config.json

Reply:
I don't understand this, but it's not important. Maybe you're reading from a slow drive? Anyways, if you think we should cache, let's cache.